jbpm-services/jbpm-executor/src/test/java/org/jbpm/executor/ExecutorQueryServiceTest.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2013 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.executor;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.UUID;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.Persistence;
+
+import org.jbpm.executor.impl.ExecutorServiceImpl;
+import org.jbpm.test.util.ExecutorTestUtil;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.api.executor.CommandContext;
+import org.kie.api.executor.ExecutorQueryService;
+import org.kie.api.executor.ExecutorService;
+import org.kie.api.executor.RequestInfo;
+
+import bitronix.tm.resource.jdbc.PoolingDataSource;
+
+/*
+ * Test case to verify proper locking of the RequestInfo table with
+ * org.jbpm.persistence.jpa.hibernate.DisabledFollowOnLockOracle10gDialect
+ *
+ * Running the test case with default H2 database succeeds, as only one thread succeeds in obtaining the RequestInfo entity
+ *
+ * To run the test with Oracle, pass in the following system parameters:
+ *
+ * // it is expected that the test case fails with default oracle dialect:
+ * -Dmaven.hibernate.dialect=org.hibernate.dialect.Oracle10gDialect
+ * 
+ * // the test should be successful using the custom dialect
+ * -Dmaven.hibernate.dialect=org.jbpm.persistence.jpa.hibernate.DisabledFollowOnLockOracle10gDialect 
```

Comment:
Thanks, @martinweiler. Just one last question - the issue was with the DisabledFollowOnLockOracle10gDialect not working properly, but Oracle10gDialect working as intended? Because the JavaDoc says that it should fail with org.hibernate.dialect.Oracle10gDialect. Is this still true?

Reply:
org.hibernate.dialect.Oracle10gDialect is not working as intended, as it does not lock a job properly. You will see 'HHH000444: Encountered request for locking however dialect reports that database prefers locking be done in a separate select' WARN messages, and the test will fail as the job is executed on both threads.

Reply:
Then I see one issue with the test. When we run full certification matrix of jbpm, we use Oracle10gDialect which is the first to be supported afaik. Is this DisabledFollowOnLockOracle10gDialect supported too? Anyway, this means that our certification matrix will fail with the Oracle database. So do we still have a bug with default Oracle10gDialect or this use case is not supported with the default Oracle10gDialect?

Reply:
I guess then we need to disable this test in the default test suite. See https://bugzilla.redhat.com/show_bug.cgi?id=1234592 as well as https://hibernate.atlassian.net/browse/HHH-1168?focusedCommentId=48846 for background information on why this custom dialect was necessary to begin with.

Reply:
@martinweiler Are you going to disable the test in this PR then?